test/with_dummyserver/test_connectionpool.py

```diff
@@ -715,6 +715,11 @@ def test_percent_encode_invalid_target_chars(self):
             r = pool.request("GET", "/echo_params?q=\r&k=\n \n")
             assert r.data == b"[('k', '\\n \\n'), ('q', '\\r')]"
 
+    def test_percent_encode_fragment(self):
+        with HTTPConnectionPool(self.host, self.port) as pool:
+            r = pool.request("GET", "/echo_params?q=1#fragment")
+            assert r.data == b"[('q', '1#fragment')]"
```

Comment:
Shouldn't this be `q=1` without the fragment? Maybe the `/echo_params` endpoint isn't the best choice for this unit test.

Reply:
Should we send the fragment identifier, though?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Identifying_resources_on_the_Web#Fragment says "the part after the #, also known as fragment identifier, is never sent to the server with the request."

As will all URL questions, this appears to be quite a complicated subject, because it looks like there are special cases with redirects, for example. If only we had an URL expert available... :) cc @sigmavirus24

Reply:
@sethmlarson I'm now using `/echo_uri` here.

Reply:
Resolving as this was fixed in #1732.